### PR TITLE
Uniform mesh update for reaction rate calculations of non-uniform assemblies

### PR DESCRIPTION
**Pull request by jakehader**
## Description

Update the uniform mesh converter to handle the reaction rate calculations and to remove the overly restrictive power conversion check. This allows the following block-level parameters for non uniform assemblies: `["rateCap", "rateFis", "rateProdN2n", "rateProdFis", "rateAbs"]`

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/reactor/converters/uniformMesh.py [hunk 1]
```diff
@@ -732,6 +756,30 @@ def _mapStateFromReactorToOther(
...
+            # Checks if the block has a multi-group flux defined and if it
+            # does not then this will skip the reaction rate calculation. This
+            # is captured by the TypeError, due to a `NoneType` divide by float
+            # error.
+            try:
+                b.getMgFlux()
+            except TypeError:
+                continue
+
+            print(b.getMgFlux())
```
Comment (albeanth) [inline armi/reactor/converters/uniformMesh.py]: Did you mean to keep these two `print` statements?

Comment (jakehader) [reply armi/reactor/converters/uniformMesh.py]: Should be removed now.

Comment (albeanth) [reply armi/reactor/converters/uniformMesh.py]: https://github.com/terrapower/armi/pull/919/commits/d30112ac4f8dc981e37d179e7fcb3f959764ce5b

File: armi/reactor/converters/uniformMesh.py [hunk 2]
```diff
@@ -537,6 +540,20 @@ def setAssemblyStateFromOverlaps(
...
+            if calcReactionRates:
+                core = sourceAssembly.getAncestor(lambda c: isinstance(c, Core))
+                if core is not None:
+                    UniformMeshGeometryConverter._calculateReactionRates(
+                        lib=core.lib, keff=core.p.keff, assem=destinationAssembly
+                    )
+                else:
+                    runLog.warning(
+                        f"Reaction rates requested for {destinationAssembly}, but no core object exists. This calculation "
+                        "will be skipped."
```
Comment (albeanth) [inline armi/reactor/converters/uniformMesh.py:554]: Will we get a zillion of these warnings? If so, do we want to add a `single = True` ?

**Comment by albeanth** [comment]
Just have one more suggestion to consider. Let's make sure that this PR doesn't break any existing uniform mesh capabilities with `detailedAxialExpansion: True`. This may be covered by unit tests, but I'm not sure off the top of my head.

File: armi/reactor/converters/uniformMesh.py (hunk 2 above)
Comment (jakehader) [reply armi/reactor/converters/uniformMesh.py:554]: We can add single = True for sure!

**Comment by jakehader** [comment]
> Just have one more suggestion to consider. Let's make sure that this PR doesn't break any existing uniform mesh capabilities with `detailedAxialExpansion: True`. This may be covered by unit tests, but I'm not sure off the top of my head.

Yes, this will work with an without detailedAxialExpansion set to True:

image

The only quirk here is that if `detailedAxialExpansion: True` _and_ we have `cs["nonUniformAssemFlags"]` then it will favor converting the reactor with `cs["nonUniformAssemFlags"]` in the following `def convert` method. I will add a settings validation that checks for this for the user.

image

**Comment by albeanth** [comment]
Awesome. The concern I had was in `UniformMeshGeometryConverter::convert`. The settings validator should cover that concern nicely.

**Pull request merged**